aspnet/mvc/models/validation.rst

```diff
@@ -1,8 +1,196 @@
+
+There may be intances where you need more features than built-in attributes provide. For those times, you can create custom validation attributes to apply to model properties.  
+
+Model State
+-----------
+`Model state <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Controller/index.html#prop-Microsoft.AspNet.Mvc.Controller.ModelState>`_ is a record of what data was bound to your model and the result of running model validation. If you have some logic that can produce a validation error, add that error to the model state's collection of errors by calling ``AddModelStateError`` as demonstrated below. Doing so will allow the the view to display the error.
+
+.. literalinclude:: validation/sample/MoviesController.cs
+   :language: c#
+   :lines: 18-39   
+   :emphasize-lines: 26, 27
+   :dedent: 8
+   
+After model binding and validation are complete, you may want to repeat parts of it. For example, a user may have entered text in a field expecting an integer. You can invoke validation again with the ``TryValidateModel`` method. 
+
+MVC will continue validating fields until reaches the maximum number of errors (200). You can configure this number by inserting the following code into the ``ConfigureServices`` method in the ``Startup.cs`` file:
+
+.. literalinclude:: validation/sample/Startup.cs
+   :language: c#
+   :lines: 5-14
+   :dedent: 4
+  
+Custom validation
+-----------------
+Data annotations work for most validation needs. However, some validation rules are specific to your business, as they're not just generic data validation such as ensuring a field is required or that it conforms to a range of values. For these scenarios, custom validation attributes are a great solution. Creating your own custom validation attributes in MVC is easy. Just inherit from the ``ValidationAttribute``, and override the ``IsValid`` method. The ``IsValid`` method accepts two parameters, the first is an object named `value` and the second is a ``ValidationContext`` object named `validationContext`. `Value` refers to the actual value from the field that your custom validator is validating. 
+
+In the following sample, a business rule that states that users may not set the genre to `Classic` for a movie released after 1960. The ``[ClassicMovie]`` attribute checks the genre first, and if it is a classic, then it checks the release date to see that it is later than 1960. If it is released after 1960, validation fails. The attribute accepts an integer parameter representing the year that you can use to validate data. You can capture the value of the parameter in the attribute's constructor, as shown here:
+					 
+.. literalinclude:: validation/sample/ClassicMovieAttribute.cs
+   :language: c#
+   :lines: 11-37
+   :dedent: 4
```

Comment:
Remove the `else` block because it doesn't add anything. Just fall through to the final `Success`.